### ISSUE TITLE
MSBuild Logger Path are not correctly quoted

### ISSUE THREAD

**Issue by gep13** [cake-build/cake#1546]
As a result, when a path with a space is used, the call to MSBuild does not work.

This is demonstrated here:

```sh
========================================
Build
========================================
Executing task: Build
Building src/Cake.DotNetCoreEf.sln
Executing: "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/MSBuild/15.0/Bin/amd64/MSBuild.exe" /m /v:n
ormal /p:Configuration="Release" /p:Platform="Any CPU" /p:TreatWarningsAsErrors=true /target:Build /logger:XmlFileLogger
,C:/Visual Studio 2017/Projects/Cake.DotNetCoreEf/tools/MSBuild.Extension.Pack/tools/net40/MSBuild.ExtensionPack.Loggers
.dll;logfile="BuildArtifacts/MsBuild.log";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8 "C:/Visual Studio 2
017/Projects/Cake.DotNetCoreEf/src/Cake.DotNetCoreEf.sln"
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1008: Only one project can be specified.
Switch: 2017/Projects/Cake.DotNetCoreEf/tools/MSBuild.Extension.Pack/tools/net40/MSBuild.ExtensionPack.Loggers.dll;logfi
le=BuildArtifacts/MsBuild.log;invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8

For switch syntax, type "MSBuild /help"
An error occurred when executing task 'Build'.
```

**Comment by gep13** [comment]
This was raised here: https://github.com/cake-contrib/Cake.DotNetCoreEf/issues/18

**Comment by gep13** [comment]
This is the affected code:

https://github.com/cake-build/cake/blob/1eb28cf18637a1316c8f10327b9d70b9975507ea/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs#L164

**Comment by devlead** [comment]
@gep13 I'll look at sorting this and get it out as an hotfix.

**Comment by devlead** [comment]
Fixed by #1558
